hudi-common/src/test/java/org/apache/hudi/metadata/TestMetadataPartitionType.java

```diff
@@ -0,0 +1,122 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.metadata;
+
+import org.apache.hudi.common.config.HoodieMetadataConfig;
+import org.apache.hudi.common.model.HoodieFunctionalIndexMetadata;
+import org.apache.hudi.common.table.HoodieTableConfig;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.Option;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests for {@link MetadataPartitionType}.
+ */
+public class TestMetadataPartitionType {
+
+  @Test
+  public void testPartitionEnabledByConfigOnly() {
+    HoodieTableMetaClient metaClient = Mockito.mock(HoodieTableMetaClient.class);
+    HoodieTableConfig tableConfig = Mockito.mock(HoodieTableConfig.class);
+
+    // Simulate the configuration enabling FILES but the meta client not having it available (yet to initialize files partition)
+    Mockito.when(metaClient.getTableConfig()).thenReturn(tableConfig);
+    Mockito.when(tableConfig.isMetadataPartitionAvailable(MetadataPartitionType.FILES)).thenReturn(false);
+    Mockito.when(metaClient.getFunctionalIndexMetadata()).thenReturn(Option.empty());
+    HoodieMetadataConfig metadataConfig = HoodieMetadataConfig.newBuilder().enable(true).build();
+
+    List<MetadataPartitionType> enabledPartitions = MetadataPartitionType.getEnabledPartitions(metadataConfig, metaClient);
+
+    // Verify FILES is enabled due to config
+    assertEquals(1, enabledPartitions.size(), "Only one partition should be enabled");
+    assertTrue(enabledPartitions.contains(MetadataPartitionType.FILES), "FILES should be enabled by config");
+  }
+
+  @Test
+  public void testPartitionAvailableByMetaClientOnly() {
+    HoodieTableMetaClient metaClient = Mockito.mock(HoodieTableMetaClient.class);
+    HoodieTableConfig tableConfig = Mockito.mock(HoodieTableConfig.class);
+
+    // Simulate the meta client having RECORD_INDEX available but config not enabling it
+    Mockito.when(metaClient.getTableConfig()).thenReturn(tableConfig);
+    Mockito.when(tableConfig.isMetadataPartitionAvailable(MetadataPartitionType.FILES)).thenReturn(true);
```

Comment:
So the meta config speified by write config does not override the config from table config metadata, is that the case? Then how can a user disable this index type once they have enabled it.

Reply:
Looks like once the user enable the metadata table with initial index type set up, they can never change it again unless the disable the whole metadata table functionality. That might need to be improved.

Reply:
I think the write config can turn off the metadata table during ingestion.  Isn't that the case?

Reply:
yeah, I mean there is no way to disable a very specific index type separately.

Reply:
nice info, so we are cool here.